src/openforms/js/components/admin/form_design/registrations/objectsapi/V2ConfigFields.js

```diff
@@ -47,20 +52,38 @@ const V2ConfigFields = ({index, name, schema, formData, onFieldChange, onChange}
     }
   }, []);
 
-  const clearMappingOnChange = message => {
+  // Resetting values if necessary:
+  useUpdateEffect(() => {
```

Comment:
We don't want to run the hook on mount, i.e. when switching from the variables tab back to the registration one.